module/data/item/spell.mjs

```diff
@@ -157,7 +157,9 @@ export default class SpellData extends ItemDataModel.mixin(
 
   /** @inheritdoc */
   get _typeAbilityMod() {
-    return this.parent?.actor?.system.attributes.spellcasting || "int";
+    return this.parent?.actor?.spellcastingClasses[this.sourceClass]?.spellcasting.ability
+      ?? this.parent?.actor?.system.attributes.spellcasting
```

Comment:
Doesn't this assume that the actor's data model has `attributes`? That may not be the case for a module subtype.